feature_engine/encoding/one_hot.py

```diff
@@ -302,3 +361,17 @@ def _add_new_feature_names(self, feature_names) -> List:
         feature_names = [f for f in feature_names if f not in self.variables_]
 
         return feature_names
+
+    def _check_custom_categories_in_dataset(self, X: pd.DataFrame) -> None:
+        """
+        Raise an error if user entered categories in custom_categories that do
+        not exist within dataset.
+
+        """
+        for var, categories in self.custom_categories.items():
+            unique_values = set(X[var].unique())
+            if not set(categories).issubset(unique_values):
```

Comment:
what if, the user wants a dummy for a category that is not present in the train set? should we contemplate this or is too unlikely?

Reply:
When someone is OHE-ing a category that is not present in the training set, are they putting it into a "catch-all" bucket, e.g. `unlearned_categories`? Is this common?

I think it's unlikely and defeats the purpose of the `custom_categories` param. I see `custom_categories` being used to protect against unlearned categories.

Reply:
Here is an example: 

I work in a car insurance company. My historical data has 3 types of fuel: gas, petrol, diesel. But, I know that electric vehicles are coming. So I want an additional category: electric. 

In the above scenario, it might help if we allow the additional bespoke category.

On the other hand, a model can't be train in a category that does not exist, so the results would be unreliable. Maybe we don't allow them for now?